zigate/responses.py

```diff
@@ -715,26 +715,20 @@ class R804E(Response):
 
     def decode(self):
         Response.decode(self)
-        additionnal = self.data.pop('additionnal')
+        LOGGER.debug("R804E data=%s", self.data)
+        additional = self.data.pop('additional')
         neighbours = []
         for i in range(self.data['count']):
-            neighbour, additionnal = self._decode('!HQQBBB',
+            if len(additional) < 3:
+                break
+            neighbour, additional = self._decode('!HQQBBB',
                                                   ['addr', 'extended_panid', 'ieee', 'depth', 'lqi', 'bit_field'],
-                                                  additionnal)
+                                                  additional)
             neighbours.append(neighbour)
         self.data['neighbours'] = neighbours
-        if additionnal:
-            self.data['addr'] = struct.unpack('!H', additionnal)[0]
+        if len(additional) == 2:
```

Comment:
This could probably be changed to `len(additional) >= 2` for potentially better forward compatibility.

Reply:
Done.